flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/operations/MergeTableLikeUtil.java

```diff
@@ -426,6 +433,14 @@ private void appendDerivedColumns(
                         }
                     }
 
+                    Set<String> physicalFieldNames = physicalFieldNamesToTypes.keySet();
+                    Set<String> metadataFieldNames = metadataFieldNamesToTypes.keySet();
+                    final Set<String> result = new LinkedHashSet<>(physicalFieldNames);
+                    result.retainAll(metadataFieldNames);
+                    if (!result.isEmpty()) {
+                        throw new ValidationException(
+                                "A field name conflict exists between a field of the regular type and a field of the Metadata type.");
+                    }
```

Comment:
Hi @wenlong88 
Thanks for your review and comments, I think we can adjust the determination of whether a SqlRegularColumn type field is duplicated with other types of fields to check if a new column is duplicated when put into the column at the end of this function.
But accessibleFieldNamesToTypes heavy logic to this place, I think it is necessary, because, when creating computecolcolumn if metadata columns in the former, repetitive physical column in the bottom, In the use of accessibleFieldNamesToTypes putAll, metadata list field covers physical column, if the user want, according to the calculated regularColumn you don't need to do so.